File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val&, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
    ListNode dummy = new ListNode(-1); 
    ListNode curr=dummy; 
    int carry=0; 

    while(l1!=null || l2!=null){

    int v1= l1!=null? l1.val : 0; 
    int v2= l2!=null ?l2.val : 0; 

    int sum= v1+v2+carry ;
    carry = sum/10; 

    curr.next= new ListNode(sum%10); 
   
    if(l1!=null)
    l1=l1.next; 
    
    if(l2!=null)
    l2=l2.next;

    curr=curr.next; 
    
    if(carry > 0){
            curr.next = new ListNode(carry);
        }
        
    }
    return dummy.next; 
}

}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """

        dummy = ListNode(-1)
        curr = dummy
        carry = 0 

        while l1 or l2: 
            v1 = l1.val if l1 else 0 
            v2 = l2.val if l2 else 0 

            currsum = v1+v2 + carry 
            carry = currsum/10                   # 12/10 = 1
            curr.next = ListNode(currsum%10)              # 12%10 = 2
            if l1: 
                l1 = l1.next 
            if l2 : 
                l2= l2.next 
            
            curr= curr.next
            
            if carry >0 : 
                curr.next = ListNode(carry) 

        return dummy.next
            